diff --git a/externals/coda-oss/modules/c++/std/include/std/type_traits b/externals/coda-oss/modules/c++/std/include/std/type_traits
index 519877b6..4c3f6454 100644
--- a/externals/coda-oss/modules/c++/std/include/std/type_traits
+++ b/externals/coda-oss/modules/c++/std/include/std/type_traits
@@ -28,7 +28,11 @@
 #ifndef CODA_OSS_NO_is_trivially_copyable
 	// https://stackoverflow.com/a/31798726/8877
 	// workaround missing "is_trivially_copyable" in g++ < 5.0
-	#if defined(__GNUC__) && (__GNUC__ < 5)
+	// later note: to be compilable with clang and thus instrumentable
+	// for taint tracking purposes, we need to be cognizant that non-GNU
+	// C and C++ compilers always define __GNUC__ < 5.
+	// c.f. https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers
+	#if defined(__GNUC__) && (__GNUC__ < 5) && !defined(__clang__)
 		#include "coda_oss/type_traits.h"
 		#define CODA_OSS_NO_is_trivially_copyable 0  // *need* our own
 	#else
